WIP: feature/10upui-plugin

devinle opened the pull request:
**WIP**
The following PR is a first pass at updating our scaffold to work with the up and coming UI plugin. This is very early stages, and includes several updates to work across the theme to the plugin. 

**Related Repo:** 
https://github.com/10up/wp-scaffold-ui 

**Related Branch:** 
https://github.com/10up/wp-scaffold-ui/pull/9

**Notes**
1. Activate the 10up theme in the main WP Scaffold repo
2. Clone the plugin WP Scaffold UI and activate in the plugins directory
3. In the plugins/wp-scaffold-ui directory, you will notice a components directory. This is where we will store each new component we create. Each component would have an index.php file where the output lives. Along with a src directory for any CSS or JS work.
4. The CSS will be set to output into the dist directory as a standalone CSS prod file. This, so an Engineer can choose individual files to include in their theme, rather than a full stylesheet.
5. CSS classes should be pluggable by theme.json. In the case of the button, we define each CSS declaration using the main theme.json property name. Secondary to this, we also provide a fallback to a foundational setting stored within the plugin file itself. This foundational style is intentionally branded to align with 10up. This provides a fallback to quickly prototype, before adding any theme specific CSS settings.
6. Also in the plugin are global tokens. You can find these here: assets/css/frontend/tokens/*.css. These tokens are all prefixed with —tui (10upUI) for now. Happy to discuss this further.
7. The theme.json settings I’ve provided are trivial. Simply for testing and presentation only. We’d likely start by defining a palette–the core colors our theme would be using.
8. From there, we can map to align with our plugin’s button component as you will see under the custom object in the theme.json. These settings will be parsed and will align with our primary CSS var declarations made in the plugin button.css file. And, as mentioned previously, if not provided via theme.json, it will fallback to our plugin’s foundational settings.
9. I’ve created a kitchen sink template for now, to display the usage options for the Button component. This makes use of the get_component method found in our plugin. It works identical to the get_template_part, but allows us to pull in partials from our plugin. Setup a new page in Admin, and assign the template to be the Button Kitchen Sink. Once done, you should be able to visit the page, and see the 2 variants of the button output to the page. Provided you have the theme.json file setup correctly, you should see those styles inherited immediately. By removing the theme.json settings, you will see it fallback to the foundational styles within plugin. Currently, the plugin is enqueuing its foundational styles, but the idea is, we wouldn’t assume this, and would allow an Engineer to pick and choose what component styles they would like to be included in their final build. For now though, these have a very small footprint and I wanted to include these as a demo.
10. Obviously CSS Propery naming patterns will be key with this approach–definitely open for discussion here as well.
11. To make this partial available to Gutenberg, I’ve created a directory inside of the 10up-theme/includes/blocks directory called tui-button (again, naming up for grabs). The index.css file within this block can import the button style from the plugin directory directly. But for now, we have the plugin enqueuing it.
12. I also made updates to the main wp-content/package.json file to run the watches on the theme and plugins. This, in the scripts section. As well, I added the plugin to the workspaces entry in the same package.json file.
13. Within the 10up-theme, I added the tui-button entry under the 10up-toolkit in order to include it in the build.
14. Within the 10up-theme/includes/blocks directory I included the tui-button block folder. This is how the partial from our plugin becomes available to Gutenberg. Here you will see the markup.php file simply pulls in the component, and outputs it. For now. We may want to discuss how exactly this works. But this allows us to use the single source of truth. And it will receive an attributes array that should align with the args that appear as options in our actual block partial found in the plugin. The index.css file simply pulls in the button CSS file from the UI plugin as well.
15. Now I run an npm install in the root of the wp-content directory, then an npm run watch to watch the theme and the plugin. I added the hot flag to work with fast refresh.
16. I added a namepsace reference to the scaffold package.json file found in the theme root.
17. I added an import of the button styles to the frontend.js file in the theme for hot reloading
18. I added the SCRIPT_DEBUG constant to wp-config as true and also WP_DEBUG for good measure

**TODO:**
I would like to create several more variants of the Button. As well, the gutenberg implementation is still lack lustre, and needs to include the plugin styles. Currently, I'm simply outputting a Rich Text input to update the label on the button. I would like to spend time making this more cohesive with the front end.

dustinrue committed:
fixes a small typo in the readme

fabiankaegy committed:
Merge pull request #98 from 10up/fix/typo

fixes a small typo in the readme

fabiankaegy committed:
fix move get_block_wrapper_attributes call into `markup.php` file

fabiankaegy committed:
fix remove reference of non existent variable class_name in block markup

fabiankaegy committed:
Merge pull request #108 from 10up/fix/escaping-of-get-wrapper-attributes

Fix: Escaping of block wrapper attributes function

Commit:
First pass at UI plugin updates

fabiankaegy commented on themes/10up-theme/includes/blocks.php:
```diff
@@ -69,6 +69,7 @@ function register_theme_blocks() {
 
 					// create helpful variables that will be accessible in markup.php file
 					$context = $block->context;
+					$wrapper_attributes = wp_kses_post( get_block_wrapper_attributes() );
```
@devinle We actually had this here previously but decided to move the `get_block_wrapper_attributes` into the actual markup.php files because you need to be able to pass properties to the `get_block_wrapper_attributes` function. 

Also it should not be escaped in that way. See #108

devinle replied:
Done!

fabiankaegy commented on themes/10up-theme/includes/blocks/tui-button/edit.js:
```diff
@@ -0,0 +1,42 @@
+				value={label}
+				onChange={(label) => setAttributes({ label })}
+			/>
+			<ServerSideRender
```
I don't think that we should use this approach here. The editorial experience should be inline. If that means that for the beginning we need to maintain both the frontend and editor js in the scaffold UI plugin I think that is still worth it because of the better editorial experience.

devinle replied:
Totally agree! I haven't worked on the editing experience. I've mostly done a test to see that the partial can be used the same way as we would in a template file. Editing experience needs to be cohesive. THANKS!

Commit:
Moves get_block_wrapper_attributes to the markup file

Commit:
Fixed typo

Commit:
Remove console test

Commit:
Moves to wp_kses_post

Commit:
Making colors more obvious from theme.json for dev purposes

joesnellpdx committed:
Update sink page output
- see todos

devinle committed:
Merge pull request #111 from 10up/feature/10upui-plugin-button-updates

Update sink page output

Commit:
Adds json copy and kitchen sink variants

joesnellpdx committed:
Update initial theme.json variables refactor

joesnellpdx committed:
Fix color palatte variables

joesnellpdx committed:
Add custom styles / output for core blocks
- theme.json
- button block custom style

joesnellpdx committed:
Add body styles to theme.json

joesnellpdx committed:
Update styles output - clean up

joesnellpdx committed:
theme.json updates

joesnellpdx committed:
Update decoration variable names. Update interactive classes in sink.

joesnellpdx committed:
remove devurl

joesnellpdx committed:
Updates to theme.json & button sink output
- add text button to sink page
- add outline and update active vars in theme.json

joesnellpdx committed:
Move UI Plugin core block styles to plugin

joesnellpdx committed:
Add class for vertical spacing to blocks

joesnellpdx committed:
Update comment for core block overrides

joesnellpdx committed:
Merge pull request #116 from 10up/feature/10upui-plugin--themejson-updates

WIP: Update initial theme.json variables, refactor UI plugin settings / output from theme

joesnellpdx committed:
Enalbe text color mods in core/button

Commit:
Updates ui kit namespacing and paths

Commit:
Update namespace for get_component use statements

Commit:
Adds partial prim, second, and tert outputs

Commit:
Rename to 10up-ui-kit

Commit:
Cleanup to pass is-style fully

Commit:
Adds light and dark

joesnellpdx committed:
Update to 6.0

Commit:
Adds dark and light background colors to button sink

joesnellpdx committed:
Merge pull request #126 from 10up/feature/text-color-additional-option

Adds dark and light background colors to button sink

joesnellpdx committed:
UI kit assets / namespace change

devinle committed:
Merge pull request #141 from 10up/feature/new-kit-asset-links

UI kit assets / namespace change

Commit:
Adds dark mode partial call

joesnellpdx committed:
Merge pull request #142 from 10up/feature/toggle-dark-mode

feature/toggle-dark-mode

Commit:
Quick fix / remove echo toggle for script

joesnellpdx committed:
Fix - hot reload

tlovett1 committed:
Fix ui kit plugin directory

Commit:
renames worspace path to align with plugin dir name

Commit:
Reverting renames ui-kit directory

Commit:
Removes non example.php template

Commit:
Removes older button component

Commit:
Remove admin inclusion of front end

joesnellpdx committed:
Add image ratio to theme.json

joesnellpdx committed:
Remove deleted bundle file from UI Kit

joesnellpdx committed:
Change buldle reference to correct file

joesnellpdx committed:
Ensure we are using the correct default theme.json file

joesnellpdx committed:
Use correct path to ui kit in package.json

joesnellpdx committed:
Update widths in theme.json

jacobdubail committed:
updates theme.json to v2 for compatibility with filtering

jacobdubail committed:
updates theme.json schema to 6.1

fabiankaegy committed:
Merge pull request #168 from 10up/feature/filter-theme-json

updates theme.json to v2 for compatibility with filtering

fabiankaegy commented:
Closing this for now
